node-src/tasks/prepare.ts

```diff
@@ -0,0 +1,206 @@
+
+const isValidStorybook = ({ paths, total }) =>
+  total > 0 && paths.includes('iframe.html') && paths.includes('index.html');
+
+export const validateFiles = async (ctx: Context) => {
+  ctx.fileInfo = getFileInfo(ctx, ctx.sourceDir);
+
+  if (!isValidStorybook(ctx.fileInfo) && ctx.buildLogFile) {
+    try {
+      const buildLog = readFileSync(ctx.buildLogFile, 'utf8');
+      const outputDirectory = getOutputDirectory(buildLog);
+      if (outputDirectory && outputDirectory !== ctx.sourceDir) {
+        ctx.log.warn(deviatingOutputDirectory(ctx, outputDirectory));
+        ctx.sourceDir = outputDirectory;
+        ctx.fileInfo = getFileInfo(ctx, ctx.sourceDir);
+      }
+    } catch (err) {
+      ctx.log.debug(err);
+    }
+  }
+
+  if (!isValidStorybook(ctx.fileInfo)) {
+    throw new Error(invalid(ctx).output);
+  }
+};
+
+// TODO: refactor this function
+// eslint-disable-next-line complexity
+export const traceChangedFiles = async (ctx: Context, task: Task) => {
+  if (!ctx.turboSnap || ctx.turboSnap.unavailable) return;
+  if (!ctx.git.changedFiles) return;
+  if (!ctx.fileInfo?.statsPath) {
+    // If we don't know the SB version, we should assume we don't support `--stats-json`
+    const nonLegacyStatsSupported =
+      ctx.storybook?.version &&
+      semver.gte(semver.coerce(ctx.storybook.version) || '0.0.0', '8.0.0');
+
+    ctx.turboSnap.bailReason = { missingStatsFile: true };
+    throw new Error(missingStatsFile({ legacy: !nonLegacyStatsSupported }));
+  }
+
+  transitionTo(tracing)(ctx, task);
+
+  const { statsPath } = ctx.fileInfo;
+  const { changedFiles } = ctx.git;
+
+  try {
+    const onlyStoryFiles = await turbosnap.traceChangedFiles(ctx);
+    if (onlyStoryFiles) {
+      // Escape special characters in the filename so it does not conflict with picomatch
+      ctx.onlyStoryFiles = Object.keys(onlyStoryFiles).map((key) =>
+        key.replaceAll(SPECIAL_CHARS_REGEXP, String.raw`\$1`)
+      );
+
+      if (!ctx.options.interactive) {
+        if (!ctx.options.traceChanged) {
+          ctx.log.info(
+            `Found affected story files:\n${Object.entries(onlyStoryFiles)
+              .flatMap(([id, files]) => files.map((f) => `  ${f} [${id}]`))
+              .join('\n')}`
+          );
+        }
+        if (ctx.untracedFiles && ctx.untracedFiles.length > 0) {
+          ctx.log.info(
+            `Encountered ${ctx.untracedFiles.length} untraced files:\n${ctx.untracedFiles
+              .map((f) => `  ${f}`)
+              .join('\n')}`
+          );
+        }
+      }
+      transitionTo(traced)(ctx, task);
+    } else {
+      transitionTo(bailed)(ctx, task);
+    }
+  } catch (err) {
+    if (!ctx.options.interactive) {
+      ctx.log.info('Failed to retrieve dependent story files', { statsPath, changedFiles, err });
+    }
+    throw rewriteErrorMessage(err, `Could not retrieve dependent story files.\n${err.message}`);
+  }
+};
+
+export const calculateFileHashes = async (ctx: Context, task: Task) => {
```

Comment:
```suggestion
export async function calculateFileHashes (ctx: Context, task: Task) {
```
(to make consistent with the rest of the file)